packages/frontend/src/components/Explorer/ExportGsheets/index.tsx

```diff
@@ -26,49 +30,78 @@ const ExportGsheets: FC<ExportGsheetProps> = memo(
-                            <Spinner
-                                className="bp4-icon bp4-icon-error"
-                                size={16}
-                            />
-                        ),
-                        key: 'exporting-gsheets',
-                        timeout: 0,
-                    });
-                },
-                onSuccess: (scheduledCsvResponse) => {
-                    pollCsvFileUrl(scheduledCsvResponse)
-                        .then((url) => {
-                            if (url) window.open(url, '_blank'); //Open in new tab
-                            AppToaster.dismiss('exporting-gsheets');
-                        })
-                        .catch((error) => {
-                            AppToaster.dismiss('exporting-gsheets');
-
-                            showToastError({
-                                title: `Unable to upload Google Sheets`,
-                                subtitle: error?.error?.message,
-                            });
-                        });
-                },
-                onError: (error: { error: Error }) => {
-                    AppToaster.dismiss('exporting-gsheets');
-
+        // TODO: move to own hook
+        const {
+            mutateAsync: startGoogleSheetExport,
+            data: startGoogleSheetExportData,
+            reset: resetStartGoogleSheetExport,
+        } = useMutation(['google-sheets-start'], getGsheetLink, {
+            onMutate: () => {
+                showToast({
+                    title: 'Exporting Google Sheets',
+                    subtitle: 'This may take a few minutes...',
+                    icon: (
+                        <Spinner
+                            className="bp4-icon bp4-icon-error"
+                            size={16}
+                        />
+                    ),
+                    key: 'exporting-gsheets',
+                    timeout: 0,
+                });
+            },
+            onError: (error: { error: Error }) => {
+                AppToaster.dismiss('exporting-gsheets');
+                showToastError({
+                    title: `Unable to upload to Google Sheets`,
+                    subtitle: error?.error?.message,
+                });
+            },
+        });
+        // TODO: move to own hook
+        const { data: exportGoogleSheetData } = useQuery<
+            ApiDownloadCsv | undefined,
+            ApiError
+        >({
+            queryKey: [`google-sheets`],
+            queryFn: () =>
+                startGoogleSheetExportData
+                    ? getCsvFileUrl(startGoogleSheetExportData)
+                    : Promise.reject({
+                          error: new Error(
+                              "Couldn't create scheduler job for google sheets export",
+                          ),
+                      }),
+            retry: (failureCount) => {
+                if (failureCount === 5) {
+                    resetStartGoogleSheetExport();
```

Comment:
after 5 failures, reset the data on first action so this doesn't refetch anymore

Reply:
https://github.com/lightdash/lightdash/assets/7611706/275dfc8a-2f5f-441e-97d8-65e0c3fb09d2

Reply:
Cool! Is 5 probably enough?